core/src/oauth/credential.rs

```diff
@@ -0,0 +1,113 @@
+use core::fmt;
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::oauth::store::OAuthStore;
+use crate::utils::{self, ParseError};
+use anyhow::Result;
+
+#[derive(Debug, Clone, Copy, Serialize, PartialEq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum CredentialProvider {
+    Snowflake,
```

Comment:
I don't think we want that

Reply:
If we want to add a new DB type we shouldn't need to touch `oauth` code at all I think

Reply:
Considering everything else we'll probably need to do to support another db, i think it's not a big deal.
I personally feel that there is some value to be able to see what each row is, especially as the content of the json is encrypted.
Do you have strong opinions on this or can I merge like that ?

Reply:
> Considering everything else we'll probably need to do to support another db, i think it's not a compelling argument.

In my view, not having to deploy an extra service to add an enum value we don't use anywhere is a good enough reason not to add this.

I guess if you really want to I can live with it, but I don't see the value in making this a required part of the API (the credentials are opaque anyway and we don't really "look" at the rows). In my opinion, less is more when it comes to APIs.

Feel free to merge as is of course, you already have an approval

Reply:
I'll keep it, to give some intent for it, as well as potential future uses (analytics, debug etc..).
Hopefully you'll be able to live with it :)

Reply:
Def wont lose sleep over it

Reply:
(Note that we’ll be able to debug the same thing easily anyway since connectors db will hold both secret id and connector provider)